### DISCUSSION TITLE
A map of the Spring-Batch repository. Please share feedback/suggestions!

### DISCUSSION THREAD

**Discussion by alexkarezin** [spring-projects/spring-batch#4435] [Ideas]
I would like to contribute to project documentation. This set of diagrams describes modules, dependencies, class hierarchy,... intending to help new developers to contribute faster. Generated from repository code and updated on schedule. Any critique of the content you can provide? Can I change/improve something? Is it worth linking from the repository docs? 
https://sourcespy.com/github/springprojectsspringbatch/
Thank you!!

**Comment by alexkarezin** [comment]
@fmbenhassine ...I don't mean to be a pest. Any chance you can take a quick look? It would mean a lot. Thx!!!

**Comment by fmbenhassine** [comment]
Thank you for sharing this with us! Interesting tool! 

> Any critique of the content you can provide? Can I change/improve something?

A couple notes:

* I don't see any version related to the current report. I guess what you shared is for the `main` branch. Are you planning to make reports versioned? This would be interesting for example to check which dependencies have changed between two versions.
* The module dependencies page shows a `null` in the title, here is an example for [spring-batch-test](https://sourcespy.com/github/springprojectsspringbatch/xx-mdependencies-org_springframework_batch__org_springframework_batch_spring-batch-test.html)
* The [documents page](https://sourcespy.com/github/springprojectsspringbatch/xx-odocuments-.html) should also include asciidocs as well (I think).
* From the main page, in the `Core Components` section, whatever module I choose, the title of the next page is `spring-batch: test: org.springframework.batch:spring-batch-XXX`. The `test` is misleading, even though the module name in the end is correct

> Is it worth linking from the repository docs?

We do not link community projects from the reference documentation nor the README, but this is the best place to share idea and projects. I will keep this discussion open for the community to see and share feedback with you.
